test/language/expressions/assignment/destructuring/obj-prop-nested-array-yield-expr.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+    When a `yield` token appears within the Initializer of a nested
+    destructuring assignment and within a generator function body, it should
+    behave as a YieldExpression.
+es6id: 12.14.5.4
```

Comment:
I'm not sure where this description is coming from --- It's a syntax error trying to parse `[x = yield]` as an `AssignmentPattern[Yield]`